feelslikemonday/app/src/main/java/com/example/feelslikemonday/ui/sendrequest/SendRequestFragment.java

```diff
@@ -94,72 +99,99 @@ public void onClick(View view) {
-                }else{
+                }
+                else if(recipientUsername.matches(myUserID) ){
+                    Toast toast = Toast.makeText(getActivity(), "This is your username, please input another one", Toast.LENGTH_LONG);
+                    toast.setGravity(Gravity.CENTER| Gravity.CENTER_HORIZONTAL, 0, 0);
+                    toast.show();
+                    usernameEditText.setText("");
+                }
+                else{
                     DAO = FollowRequestDAO.getInstance();
+                    followPermissionDAO = FollowPermissionDAO.getInstance();
 
-                    //Check Input User exists in the database
-                    DAO.get(recipientUsername, new FollowRequestCallback(){
+                    DAO.get(recipientUsername, new FollowRequestCallback() {
                         @Override
                         public void onCallback(FollowRequest followRequest) {
-                            if(followRequest.getRecipientUsername().equals(recipientUsername)){
-                                requesterUsernames = followRequest.getRequesterUsernames();
-                                requesterUsernames.add(myUserID);   //TODO: check you have not request twice.
-                                Toast toast = Toast.makeText(getActivity(), "Valid User", Toast.LENGTH_LONG);
-                                toast.setGravity(Gravity.CENTER| Gravity.CENTER_HORIZONTAL, 0, 0);
-                                toast.show();
-                                followRequest_ok = followRequest;
-
-                                DAO.createOrUpdate(recipientUsername, followRequest_ok, new VoidCallback() {
-                                    @Override
-                                    public void onCallback() {
-                                    }
-                                });
-                                usernameEditText.setText("");
-
-                                Toast toast1 = Toast.makeText(getActivity(), "Send Request Successfully to " + recipientUsername, Toast.LENGTH_LONG);
-                                toast1.setGravity(Gravity.CENTER| Gravity.CENTER_HORIZONTAL, 0, 0);
-                                toast1.show();
+                            if (followRequest.getRecipientUsername().equals(recipientUsername)) {
+
+                                ValidUser();
+
+                                followPermissionDAO.get(recipientUsername, new FollowPermissionCallback() {
+                                            @Override
+                                            public void onCallback(FollowPermission followPermission) {
+                                                if (followPermission.getFolloweeUsernames().contains(myUserID)) {
+                                                    AlreadyFollowedUser();
+                                                }else{
+                                                    DAO.get(recipientUsername, new FollowRequestCallback() {
```

Comment:
Can we specify what DAO is? Rename DAO to something like ____DAO?

Reply:
fixed